### ISSUE TITLE
Name conflict of "provide" methods

### ISSUE THREAD

**Issue by maxandron** [google/dagger#325]
Hi,

TL;DR: Using the same name for a "provided" method in completely different models stops dagger from generating random files

This is a bug I encountered which was incredibly annoying, I spent two whole days trying to figure it out.
I recently converted my project to Kotlin so I blamed it on them - turns out it's dagger.

The bug occurred because in module "A" I had the following method:
`@Provides @AScope fun providePresenter(model: model)`

and then in module "B" (completely different file) I had again a similar method:
`@Provides @BScope fun providePresenter(model: model)`

The main problem was the error I received:
`Error:(27, 32) error: cannot find symbol class BProvider_ProvidePresenterFactory`
And the BProvider_ProvidePresenterFactory was actually not even generated.

Hope it doesn't waste any more time to anyone else -.-

**Comment by ronshapiro** [comment]
Can you reproduce it?

**Comment by maxandron** [comment]
Easily. Just use two different `@Modules` that provide something different but with the same name.
Tell me if a sample project is needed, I'll make one

**Comment by ronshapiro** [comment]
That'd be great!

**Comment by maxandron** [comment]
I tried reproducing in Java and had no success.
So it is a kotlin issue. (At least I think so)

Here is a sample project: https://github.com/maxandron/DaggerIssue325Java

Tell me if I can help further.

**Comment by maxandron** [comment]
Another two cents... the same thing seems to be happening with butterknife. This time when using the same variable name in two completely different classes in different packages

**Comment by bgogetap** [comment]
This issue has been fixed since Kotlin 1.0.2 Early Access Previews. It was a kapt issue and I believe this can be closed.
